Sources/WebSocketClient/WebSocketClient.swift

```diff
@@ -0,0 +1,685 @@
+    //         Example usage with compression enabled:
+    //             let client = WebSocketClient(host: "localhost", port: 8080, uri: "/", requestKey: "test", negotiateCompression: true)
+    //
+    // - parameters:
+    //     - host: Host name of the remote server
+    //     - port: Port number on which the remote server is listening
+    //     - uri : The "Request-URI" of the GET method, it is used to identify the endpoint of the WebSocket connection
+    //     - requestKey: The requestKey sent by client which server has to include while building it's response. This helps ensure that the server
+    //                   does not accept connections from non-WebSocket clients
+    //     - maxFrameSize : Maximum allowable frame size of WebSocket client is configured using this parameter.
+    //                      Default value is `14`.
+    //     - compressionConfig : compression configuration
+
+    public init?(host: String, port: Int, uri: String, requestKey: String,
+                 compressionConfig: CompressionConfig? = nil, maxFrameSize: Int = 14, enableSSL: Bool = false, onOpen: @escaping (Channel?) -> Void = { _ in }) {
+        self.requestKey = requestKey
+        self.host = host
+        self.port = port
+        self.uri = uri
+        self.onOpenCallback = onOpen
+        self.compressionConfig = compressionConfig
+        self.maxFrameSize = maxFrameSize
+        self.enableSSL = enableSSL
+    }
+
+    // Create a new `WebSocketClient`.
+    //
+    //
+    //         Example usage:
+    //             let client = WebSocketClient("ws://localhost:8080/chat")
+    //
+    //         // See RFC 7692 for to know more about compression negotiation
+    //         Example usage with compression enabled:
+    //             let client = WebSocketClient(host: "localhost", port: 8080, uri: "/", requestKey: "test", negotiateCompression: true)
+    //
+    // - parameters:
+    //     - url : The "Request-URl" of the GET method, it is used to identify the endpoint of the WebSocket connection
+    //     - compressionConfig : compression configuration
+
+    public init?(_ url: String, config: CompressionConfig? = nil) {
+        self.requestKey = "test"
+        let rawUrl = URL(string: url)
+        self.host = rawUrl?.host ?? "localhost"
+        self.port = rawUrl?.port ?? 8080
+        self.uri =  rawUrl?.path ?? "/"
+        self.compressionConfig = config
+        self.maxFrameSize = 24
+        self.enableSSL = (rawUrl?.scheme == "wss" || rawUrl?.scheme == "https") ? true : false
+    }
+
+    let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+
+    public var delegate: WebSocketClientDelegate? = nil
+
+    // Whether close frame is sent to server
+    var closeSent: Bool = false
+
+    // Whether the client is still alive
+    public var isConnected: Bool {
+        return self.channel?.isActive ?? false
+    }
+
+    public func connect() {
```

Comment:
We need to have an error reporting mechanism here. Can we create a  new Error type and throw it from the `catch`? Something like `WebSocketConnectionFailedError`

Reply:
done